internal/app/machined/pkg/system/services/bootkube.go

```diff
@@ -0,0 +1,210 @@
+}
+
+// nolint: gocyclo
+func generateAssets(config config.Configurator) (err error) {
+	if err = os.MkdirAll("/etc/kubernetes/manifests", 0644); err != nil {
+		return err
+	}
+
+	peerCrt, err := ioutil.ReadFile(constants.KubernetesEtcdPeerCert)
+	if err != nil {
+		return err
+	}
+	block, _ := pem.Decode(peerCrt)
+	if block == nil {
+		return errors.New("failed to decode peer certificate")
+	}
+	peer, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse client certificate")
+	}
+
+	caCrt, err := ioutil.ReadFile(constants.KubernetesEtcdCACert)
+	if err != nil {
+		return err
+	}
+	block, _ = pem.Decode(caCrt)
+	if block == nil {
+		return errors.New("failed to decode CA certificate")
+	}
+	ca, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse etcd CA certificate")
+	}
+
+	peerKey, err := ioutil.ReadFile(constants.KubernetesEtcdPeerKey)
+	if err != nil {
+		return err
+	}
+	block, _ = pem.Decode(peerKey)
+	if block == nil {
+		return errors.New("failed to peer key")
+	}
+	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse client key")
+	}
+
+	etcdServer, err := url.Parse("https://127.0.0.1:2379")
+	if err != nil {
+		return err
+	}
+
+	apiServers := []*url.URL{}
+	for _, ip := range config.Cluster().IPs() {
+		var u *url.URL
+		if u, err = url.Parse("https://" + ip + ":6443"); err != nil {
+			return err
+		}
+		apiServers = append(apiServers, u)
+	}
+	u, err := url.Parse("https://127.0.0.1:6443")
+	if err != nil {
+		return err
+	}
+	apiServers = append(apiServers, u)
+
+	_, podCIDR, err := net.ParseCIDR("10.2.0.0/16")
+	if err != nil {
+		return err
+	}
+	_, serviceCIDR, err := net.ParseCIDR("10.3.0.0/24")
+	if err != nil {
+		return err
+	}
+
+	altNames := altNamesFromURLs(apiServers)
+
+	block, _ = pem.Decode(config.Cluster().CA().Crt)
+	if block == nil {
+		return errors.New("failed to Kubernetes CA certificate")
+	}
+	k8sCA, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse Kubernetes CA certificate")
+	}
+
+	block, _ = pem.Decode(config.Cluster().CA().Key)
+	if block == nil {
+		return errors.New("failed to Kubernetes CA key")
+	}
+	k8sKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse Kubernetes key")
+	}
+
+	conf := asset.Config{
+		CACert:                 k8sCA,
+		CAPrivKey:              k8sKey,
+		EtcdCACert:             ca,
+		EtcdClientCert:         peer,
+		EtcdClientKey:          key,
+		EtcdServers:            []*url.URL{etcdServer},
+		EtcdUseTLS:             true,
+		APIServers:             apiServers,
+		APIServiceIP:           net.ParseIP("10.3.0.1"),
+		DNSServiceIP:           net.ParseIP("10.3.0.10"),
+		PodCIDR:                podCIDR,
+		ServiceCIDR:            serviceCIDR,
+		NetworkProvider:        "flannel",
```

Comment:
DefaultCNI?

Reply:
For now at least. As discussed today, we'd like to give the ability to point this somewhere remote.